src/main/java/com/redhat/devtools/lsp4ij/LanguageServerFactory.java

```diff
@@ -50,4 +51,7 @@ public interface LanguageServerFactory {
         return LanguageServer.class;
     }
 
+    @NotNull default LSPClientFeatures createClientFeatures() {

```

Comment:
I agree with you.

In short answer, it exists LSPClientFeature#getProject()
 
LSPClientFeatures is set with the LanguageServerWrapper which have the project. We need LanguageServerWrapper to support server capability  see LSPFormattingFeature.

The LanguageServerWrapper set her instance when LSPClientFeatures is created (I did that because I don't want to expose LanguageServerWrapper  in the API)

The LanguageServerWrapper hosts the Project, so LSPClientFeatures know the Project.